ktor-client/ktor-client-curl/desktop/src/io/ktor/client/engine/curl/CurlProcessor.kt

```diff
@@ -42,6 +43,11 @@ internal class CurlProcessor(coroutineContext: CoroutineContext) {
     }
 
     suspend fun executeRequest(request: CurlRequestData): CurlSuccess {
+        if (request.isUpgradeRequest && !curlProtocols.contains("ws")) {
+            @Suppress("DEPRECATION")
+            throw CurlRuntimeException("WebSockets are supported in experimental libcurl 7.86 and greater")
+        }
+
```

Comment:
It wouldn't since `curlProtocols` is a list of protocols supported by this curl installation, so if `wss` is in there `ws` would be in there to, therefore passing this check, although it could probabbly be optimized to be

```suggestion
        if (request.isUpgradeRequest && !curlProtocols.contains(request.url.substringBefore("://"))) {
            @Suppress("DEPRECATION")
            throw CurlRuntimeException("WebSockets are supported in experimental libcurl 7.86 and greater")
        }

```

Reply:
Thanks, I added the actual protocol verification.